File: Sandwitch.Client/src/viewmodels/views/viewarenal.ts
import { ViewBase } from './viewbase';
import { ViewCatalog } from './viewcatalog';
import { ViewHistorico } from './viewhistorico';

export interface ViewArenal extends ViewBase {
  Name: string;
  LastHistorico: ViewHistorico;
  Poblaciones: ViewCatalog[];
}

File: Sandwitch.Service/Sandwitch.Application/ViewModels/Views/ViewArenal.cs
namespace Sandwitch.Application.ViewModels.Views;

/// <summary>
///     Represents a <see cref="ViewArenal" /> class. Inherits <see cref="ViewBase" />
/// </summary>
public class ViewArenal : ViewBase
{   
    /// <summary>
    /// Gets or Sets <see cref="Name"/>
    /// </summary>
    public string Name { get; set; }

    /// <summary>
    /// Gets or Sets <see cref="Poblaciones"/>
    /// </summary>
    public virtual ICollection<ViewCatalog> Poblaciones { get; set; } = [];

    /// <summary>
    /// Gets or Sets <see cref="LastHistorico"/>
    /// </summary>
    public virtual ViewHistorico LastHistorico { get; set; }
}